GroupManager/App_Start/Startup.Auth.cs

```diff
@@ -1,80 +1,113 @@
-                    {
-                        // We'll inject our own issuer validation logic below.
-                        ValidateIssuer = false,
-                        NameClaimType = "name",
-                    },
-                    Notifications = new OpenIdConnectAuthenticationNotifications()
-                    {
-                        SecurityTokenValidated = OnSecurityTokenValidated,
-                        AuthorizationCodeReceived = OnAuthorizationCodeReceived,
-                        AuthenticationFailed = OnAuthenticationFailed,
-                    }
-                });
-        }
+		private Task OnAuthenticationFailed(AuthenticationFailedNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions> context)
+		{
+			// Handle any unexpected errors during sign in
+			context.OwinContext.Response.Redirect("/Error?message=" + context.Exception.Message);
+			context.HandleResponse(); // Suppress the exception
+			return Task.FromResult(0);
+		}
 
-        private Task OnAuthenticationFailed(AuthenticationFailedNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions> context)
-        {
-            // Handle any unexpected errors during sign in
-            context.OwinContext.Response.Redirect("/Error?message=" + context.Exception.Message);
-            context.HandleResponse(); // Suppress the exception
-            return Task.FromResult(0);
-        }
+		private async Task OnAuthorizationCodeReceived(AuthorizationCodeReceivedNotification context)
+		{
+			// Upon successful sign in, get & cache a token using MSAL
+			IConfidentialClientApplication confidentialClient = ConfidentialClientApplicationBuilder.Create(Globals.ClientId)
+				  .WithClientSecret(Globals.ClientSecret)
+				  .WithRedirectUri(Globals.RedirectUri)
+				  .WithAuthority(new Uri(Globals.Authority))
+				  .Build();
 
-        private async Task OnAuthorizationCodeReceived(AuthorizationCodeReceivedNotification context)
-        {
-            // Upon successful sign in, get & cache a token using MSAL
-            string userId = context.AuthenticationTicket.Identity.FindFirst(ClaimTypes.NameIdentifier).Value;
-            IConfidentialClientApplication cc = MsalAppBuilder.BuildConfidentialClientApplication();
+			/*
+			 The `MSALPerUserMemoryTokenCache` is created and hooked in the `UserTokenCache` used by `IConfidentialClientApplication`.
+			 At this point, if you inspect `ClaimsPrinciple.Current` you will notice that the Identity is still unauthenticated and it has no claims,
+			 but `MSALPerUserMemoryTokenCache` needs the claims to work properly. Because of this sync problem, we are using the constructor that
+			 receives `ClaimsPrincipal` as argument and we are getting the claims from the object `AuthorizationCodeReceivedNotification context`.
+			 This object contains the property `AuthenticationTicket.Identity`, which is a `ClaimsIdentity`, where all the props are filled
```

Comment:
>where all the props are filled [](start = 100, length = 30)

No shortcuts "props" :) "created from the token received from Azure AD and has a full set of claims"

Reply:
Fixed